kernel/src/log_reader/manifest.rs

```diff
@@ -0,0 +1,309 @@
+//! Manifest phase for log replay - processes single-part checkpoints and manifest checkpoints.
+
+use std::sync::{Arc, LazyLock};
+
+use itertools::Itertools;
+use url::Url;
+
+use crate::actions::visitors::SidecarVisitor;
+use crate::actions::{Add, Remove, Sidecar, ADD_NAME};
+use crate::actions::{REMOVE_NAME, SIDECAR_NAME};
+use crate::log_replay::ActionsBatch;
+use crate::path::ParsedLogPath;
+use crate::schema::{SchemaRef, StructField, StructType, ToSchema};
+use crate::utils::require;
+use crate::{DeltaResult, Engine, Error, FileMeta, RowVisitor};
+
+/// Phase that processes single-part checkpoint. This also treats the checkpoint as a manifest file
+/// and extracts the sidecar actions during iteration.
+#[allow(unused)]
+pub(crate) struct ManifestPhase {
```

Comment:
I don't _love_ the name `ManifestPhase`, because it doesn't tell you what it's a phase _of_. Could this just be a `CheckpointManifestProcessor`?

Reply:
I went with `CheckpointManifestReader`